src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotMatchKeywords.cs

```diff
@@ -95,28 +96,28 @@ public sealed class IdentifiersShouldNotMatchKeywordsAnalyzer : DiagnosticAnalyz
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            // TODO: Make this analyzer thread-safe.
-            //analysisContext.EnableConcurrentExecution();
-
+            analysisContext.EnableConcurrentExecution();
             analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            var keywordNamedNamespaces = new HashSet<string>();
+            analysisContext.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                var keywordNamedNamespaces = new ConcurrentDictionary<string, string>();
```

Comment:
I was meaning to say concurrent hashset but that seems like an internal Roslyn utilities data structure. Using a dictionary to store just a set of values seems weird and concurrent dictionary causes lot of locking overhead - can we use a regular hashset and just lock on its reads and writes?